src/pymatgen/io/cp2k/inputs.py

```diff
@@ -2008,11 +2008,11 @@ def from_kpoints(cls, kpoints: VaspKpoints, structure: Structure | None = None)
                 do this automatically without spglib present at execution time.
         """
         kpts = kpoints.kpts
-        weights = kpoints.kpts_weights
+        weights: Sequence[float] | None = kpoints.kpts_weights
 
         if kpoints.style == KpointsSupportedModes.Monkhorst:
-            kpt: Kpoint | float = kpts[0]
-            x, y, z = (kpt, kpt, kpt) if isinstance(kpt, (int, float)) else kpt
```

Comment:
Might need additional check on 578687a085f9867a6c6f900b763e7ebbf7329b62. Now `Kpoint` cannot be a `float | int`, but `tuple[float, float, float] | tuple[int]` only